src/microbe_stage/Spawner.cs

```diff
@@ -1,50 +1,113 @@
+            return binomialValuesCache;
+        }
+    }
 
     /// <summary>
-    ///   How much stuff spawns
+    ///   Evenly distributes the spawns in a sector.
     /// </summary>
-    /// <value>The spawn frequency.</value>
-    public int SpawnFrequency { get; set; }
+    /// <returns>Returns the relative points where stuff should spawn</returns>
+    public virtual IEnumerable<Vector3> GetSpawnPoints(float sectorDensity,
+        Random random)
+    {
+        var spawns = GetSpawnsInASector(sectorDensity, random);
+        var results = new List<Vector3>(spawns);
+        for (var i = 0; i < spawns; i++)
+        {
+            var vector = new Vector3(random.NextFloat(), 0, random.NextFloat());
+            vector *= Constants.SECTOR_SIZE;
+
+            // Check if another spawn is too close
+            if (results.Any(v => (v - vector).LengthSquared() < MinDistanceSquared))
+            {
+                i--;
+                continue;
+            }
+
+            results.Add(vector);
+        }
+
+        return results;
+    }
 
     /// <summary>
-    ///   If this is queued to be destroyed the spawn system will remove this on next update
+    ///   Instantiate the next thing.
+    ///   This is an enumerator to be able to control how many things to spawn per frame easily.
+    ///   Do not add those instances to the world, that's the spawn system's job.
     /// </summary>
-    /// <value><c>true</c> if destroy queued; otherwise, <c>false</c>.</value>
-    public bool DestroyQueued { get; set; }
+    /// <param name="location">Location the spawn system wants to spawn a thing at</param>
+    /// <returns>An enumerator which defines the instances of the object to spawn</returns>
+    public abstract IEnumerable<SpawnedRigidBody>? Instantiate(Vector3 location);
 
     /// <summary>
-    ///   Spawns the next thing. This is an enumerator to be able to control how many things to spawn per frame easily
+    ///   Used in <see cref="GetSpawnPoints"/> to determine how many spawns should occur in this sector and therefore
+    ///   the length of the list returned vector list.
     /// </summary>
-    /// <param name="worldNode">The parent node of spawned entities</param>
-    /// <param name="location">Location the spawn system wants to spawn a thing at</param>
-    /// <returns>An enumerator that on each next call spawns one thing</returns>
-    public abstract IEnumerable<ISpawned>? Spawn(Node worldNode, Vector3 location);
+    /// <returns>Returns the amount of spawns in a sector.</returns>
+    protected virtual int GetSpawnsInASector(float sectorDensity, Random random)
+    {
+        if (BinomialP == 0)
+            return 0;
+
+        var nextRandom = random.NextFloat() * sectorDensity;
+        var binomialSum = 0f;
+        var i = 0;
+        do
+        {
+            binomialSum += BinomialValues[i++];
+        }
+        while (nextRandom >= binomialSum);
+
+        return i - 1;
+    }
```

Comment:
Why this probability law in particular? I assume it at least privileges central values, since nCp is highest for p = n/2 (if I'm not mistaken).

If it is a known method, could you please add a reference in the doc? That would be super helpful in reviewing the mechanics, especially if we consider tweaking it.